### PR TITLE
tail: handle `-` as an alias for stdin

### PR DESCRIPTION
**Pull request by thomasqueirozb**

### CODE REVIEW

**Comment by sylvestre** [comment]
please add a test to make sure that this is covered! thanks

**Review by sylvestre — CHANGES_REQUESTED** [review]
needs a test, should be easy to implement, see:
https://github.com/uutils/coreutils/blob/15da98d84e9a094ea72c5f51efcc2d8aa9e9184f/tests/by-util/test_wc.rs#L41-L48

File: src/uu/tail/src/tail.rs [hunk 1]
```diff
@@ -23,12 +24,19 @@ use std::collections::VecDeque;
 use std::fmt;
 use std::fs::File;
 use std::io::{stdin, stdout, BufRead, BufReader, Read, Seek, SeekFrom, Write};
+use std::os::unix::prelude::AsRawFd;
```
Comment (sylvestre) [inline src/uu/tail/src/tail.rs:27]: and please fix this warning :)

**Comment by sylvestre** [comment]
@thomasqueirozb ping?

**Comment by thomasqueirozb** [comment]
> @thomasqueirozb ping?

pong

**Comment by thomasqueirozb** [comment]
I have been away from my computer the past week so I haven't been able to fix this. I will do this either tomorrow or the day after. Sorry for the delay

File: src/uu/tail/src/tail.rs [hunk 2]
```diff
@@ -130,25 +140,84 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
-                if !first_header {
-                    println!();
-                }
+        .unwrap_or_else(|| vec![String::from("-")]);
+
+    let multiple = files.len() > 1;
+    let mut first_header = true;
+    let mut readers: Vec<(Box<dyn BufRead>, &String)> = Vec::new();
+
+    let stdin_string = String::from("standard input");
```
Comment (sylvestre) [inline src/uu/tail/src/tail.rs:149]: the compiler is complaining that
"unused variable: `stdin_string`"

File: src/uu/tail/src/tail.rs [hunk 3]
```diff
@@ -130,25 +140,84 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+            } else {
                 println!("==> {} <==", filename);
             }
-            first_header = false;
+        }
+        first_header = false;
+
+        if use_stdin {
+            #[cfg(unix)]
+            {
```
Comment (sylvestre) [inline src/uu/tail/src/tail.rs:166]: maybe move this block into a dedicated function?

File: src/uu/tail/src/tail.rs [hunk 4]
```diff
@@ -29,6 +30,8 @@ use std::time::Duration;
 use uucore::parse_size::{parse_size, ParseSizeError};
 use uucore::ringbuffer::RingBuffer;
 
+use crate::platform::stdin_is_pipe_or_fifo;
```
Comment (sylvestre) [inline src/uu/tail/src/tail.rs:33]: could you please fix this warning?
```
unresolved import `crate::platform::stdin_is_pipe_or_fifo`
```

**Comment by thomasqueirozb** [comment]
I hate conditional compilation... I can only see those messages after the CI runs

**Comment by jfinkels** [comment]
Fixed #1323.

**Pull request merged**